src/canvas.jl

```diff
@@ -108,62 +108,49 @@ function get_canvas_dimensions_for_matrix(
     width::Int, height::Int, margin::Int, padding::Int, out_stream::Union{Nothing,IO};
     extra_rows::Int = 0, extra_cols::Int = 0
 ) where {T <: Canvas}
-    min_canvheight = ceil(Int, nrow / y_pixel_per_char(T))
-    min_canvwidth  = ceil(Int, ncol / x_pixel_per_char(T))
-    aspect_ratio = min_canvwidth / min_canvheight
+    canvheight = nrow / y_pixel_per_char(T)
+    canvwidth  = ncol / x_pixel_per_char(T)
+    aspect_ratio = canvwidth / canvheight
```

Comment:
Maybe leave a comment here for why `aspect_ratio` is calculated this way? My understanding is that we faithfully calculate the desired aspect ratio while keeping the canvas big enough to fit all the contents?

Reply:
I've added comments base on a reduced example (2x2 Matrix).